Apps/WebClient/src/ClientApp/app/components/timeline/comment.vue

```diff
@@ -4,28 +4,255 @@
+        >
+          Only you can see comments added to your medical records.
+        </b-tooltip>
+      </div>
+      <div class="comment-input pl-2">
+        <b-form @submit.prevent>
+          <b-form-textarea
+            ref="commentInput"
+            v-model="commentInput"
+            rows="1"
+            no-resize
+            :placeholder="placeholder"
+            maxlength="1000"
+          ></b-form-textarea>
+        </b-form>
+      </div>
+      <div class="d-flex comment-button px-3 flex-row">
+        <b-button
+          variant="primary"
+          :disabled="commentInput === ''"
+          class="d-flex"
+          @click="onSubmit"
+        >
+          Save
+        </b-button>
+        <div v-if="!isNewComment" class="d-flex pl-2">
+          <b-button variant="secondary" @click="onCancel">
+            Cancel
+          </b-button>
+        </div>
+      </div>
     </b-row>
-    <b-row class="px-3">
+    <b-row v-if="!isNewComment" class="px-3">
       <span> {{ formatDate(comment.createdDateTime) }} </span>
     </b-row>
   </b-col>
 </template>
 <script lang="ts">
 import Vue from "vue";
 import UserComment from "@/models/userComment";
-import { Prop, Component } from "vue-property-decorator";
+import User from "@/models/user";
+import { Getter } from "vuex-class";
+import { Prop, Component, Emit, Watch } from "vue-property-decorator";
+import {
+  faEllipsisV,
+  IconDefinition,
+  faLock
+} from "@fortawesome/free-solid-svg-icons";
+import { IUserCommentService } from "@/services/interfaces";
+import { SERVICE_IDENTIFIER } from "@/plugins/inversify";
+import container from "@/plugins/inversify.config";
 
 @Component
 export default class CommentComponent extends Vue {
-  @Prop({ default: "Default" }) comment!: UserComment;
+  @Getter("user", { namespace: "user" }) user!: User;
+  @Prop() comment!: UserComment;
+  private commentInput: string = "";
+  private commentService!: IUserCommentService;
+  private hasErrors: boolean = false;
+  private isEditMode: boolean = false;
+
+  private mounted() {
+    this.commentService = container.get<IUserCommentService>(
+      SERVICE_IDENTIFIER.UserCommentService
+    );
+  }
+
+  private get placeholder(): string {
+    if (this.isEditMode) {
+      return "Editing a comment";
+    } else {
+      return "Add a private comment";
+    }
+  }
+
+  private get isNewComment(): boolean {
+    return this.comment.id === ""
+  }
+
   private formatDate(date: Date): string {
     return new Date(Date.parse(date + "Z")).toLocaleString();
   }
+
+  private get menuIcon(): IconDefinition {
+    return faEllipsisV;
+  }
+
+  private get lockIcon(): IconDefinition {
+    return faLock;
+  }
+
+  private onSubmit(): void {
+    if (this.isEditMode) {
+      this.updateComment();
+    } else {
+      this.addComment();
+    }
+  }
+
+  private onCancel(): void {
+    this.isEditMode = false;
+  }
+
+  private addComment(): void {
+    let commentPromise = this.commentService
```

Comment:
on all events we could use a spinner or some sort of indication that the system is working. Starting before the promise, and removed after it completes

Reply:
added